pkg/sync/adapter.go

```diff
@@ -197,6 +197,23 @@ func (a Adapter) pushLocal(path string, files []string, delFiles []string, isFor
 		}
 	}
 
+	if isForcePush {
+		// If isForcePush is set, then the entire local directory contents will be copied over by CopyFile(...), thus
+		// we first need to clear the remote target folder.
+
+		err = exec.ExecuteCommand(a.Client, compInfo, []string{"rm", "-rf", syncFolder}, true, nil, nil)
+		if err != nil {
+			// This command may fail if the parent directory permissions are restrictive; since this doesn't
+			// affect our ability to delete the files _within_ the directory, we merely log it here.
```

Comment:
>  ability to delete the files _within_ the directory

can you please elaborate this line?

Reply:
I've changed the logic and updated the comment, let me know what you think now, [see my comment here](https://github.com/openshift/odo/pull/3619#discussion_r460817847).